plugins/modules/zos_replace.py

```diff
@@ -0,0 +1,646 @@
+        name generated by calling the ZOAU API. The MVS backup data set
+        recovery can be done by renaming it.
+      - If I(src) is a data set member and backup_name is not provided, the data set
+        member will be backed up to the same partitioned data set with a randomly generated
+        member name.
+    required: false
+    type: str
+  before:
+    description:
+      - If specified, only content before this match will be replaced/removed.
+      - Can be used in combination with I(after).
+    required: false
+    type: str
+  encoding:
+    description:
+      - The character set of the source I(target). L(zos_replace,./zos_replace.html)
+        requires it to be provided with correct encoding to read the content
+        of a USS file or data set. If this parameter is not provided, this
+        module assumes that USS file or data set is encoded in IBM-1047.
+      - Supported character sets rely on the charset conversion utility (iconv)
+        version; the most common character sets are supported.
+    required: false
+    type: str
+    default: IBM-1047
+  literal:
+    description:
+      - A list to allow the user to use I(before), I(after) or I(regexp) as regular str not regex.
+    type: list
+    default: []
+    choices:
+      - after
+      - before
+      - regexp
+  target:
+    description:
+      - The location can be a UNIX System Services (USS) file,
+        PS (sequential data set), member of a PDS or PDSE, PDS, PDSE.
+      - The USS file must be an absolute pathname.
+      - Generation data set (GDS) relative name of generation already
+        created. e.g. I(SOME.CREATION(-1)).
+    required: true
+    type: str
+    aliases: [ src, path, destfile ]
+  tmp_hlq:
+    description:
+      - Override the default high level qualifier (HLQ) for temporary and backup
+        datasets.
+      - The default HLQ is the Ansible user used to execute the module and if
+        that is not available, then the value C(TMPHLQ) is used.
+    required: false
+    type: str
+  regexp:
+    description:
+      - The regular expression to look for in the contents of the file.
+    required: true
+    type: str
+  replace:
+    description:
+      - The string to replace regexp matches.
+      - If not set, matches are removed entirely.
+    required: false
+    type: str
+    default: ""
+"""
+
+EXAMPLES = r"""
+- name: Replace lines on gdg and generate backup on gdg
+  zos_replace:
+    target: SOURCE.GDG(0)
+    regexp: ^(IEE132I|IEA989I|IEA888I|IEF196I|IEA000I)\s.*
+    after: ^IEE133I PENDING *
+    before: ^IEE252I DEVICE *
+    backup: True
+    backup_name: "SOURCE.GDG(+1)"
+
+- name: Replace words with backref on PDSE to delete some calls of SYSTEM
+  zos_replace:
+    target: PDS.SOURCE(MEM)
+    regexp: '^(.*?SYSTEM.*?)SYSTEM(.*)'
+    replace: '\1\2'
+    after: IEE133I PENDING *
+    before: IEF456I JOB12345 *
+
+- name: Replace using after on USS with no utf-8 characters
```

Comment:
It doesn't make any difference in the task definition the fact that the USS has no utf-8 characters so no point in mentioning. Rather you could explain what this regexp is looking for